executor/src/witgen/machines/block_machine.rs

```diff
@@ -165,10 +165,27 @@ impl<'a, T: FieldElement> Machine<'a, T> for BlockMachine<'a, T> {
         mutable_state: &'b mut MutableState<'a, 'b, T, Q>,
     ) -> HashMap<String, Vec<T>> {
         if self.data.len() < 2 * self.block_size {
-            log::warn!(
-                "Filling empty blocks with zeros, because the block machine is never used. \
+            if self.degree_range.is_unique() {
```

Comment:
We can do better here: even if we're using vadcop, a range check machine would have a unique degree. If it is not accessed, this would still prove it. 

The correct decision factor is more global, it's whether the main machine has a unique degree, see linker.

Maybe this is a sign we should have a flag for this instead of inferring.

Reply:
This code only runs for block machines though! Seems unlikely that dynamic VADCOP is enabled but the block machine has a static size.

But in general, I agree.

Reply:
Also I agree that we should do this not only for block machines, but also secondary VMs, fixed lookups, memory, ...

I think this can happen in a separate PR though. Block machines are the most crucial, because they need a default block if they can't be empty. Also, for fixed lookups, it'll depend on whether any multiplicity is non-zero, which we'll only know once we fully implemented the bus (multiplicities are only counted for `PhantomLookup`s).

Reply:
I mean, until we have an explicit flag, I could check whether, for the entire PIL, all fixed columns have the same static size. Otherwise, I assume VADCOP is supported and I can set a size to zero. 

Think that would be better than only checking the current machine, right?

Reply:
OK, done!